### PR TITLE
add the support for dumping best trajectories during training

### PR DESCRIPTION
**Pull request by yundiqian**
add the support for dumping best trajectories during training for later use. It finds the most optimal action list for individual modules/functions

the naming of policy_rewards, when put together with rewards in the CompilationResult is confusing. will have a full codebase refactor to refactor {default|policy}_reward to {|}_cost after #31 is done

### CODE REVIEW

**Comment by mtrofin** [comment]
You can reference #31 instead of "migration to executor", since the latter is an internal effort.

**Comment by yundiqian** [comment]
> 

done

File: compiler_opt/rl/local_data_collector.py [hunk 1]
```diff
@@ -118,7 +121,7 @@ def collect_data(
     """Collect data for a given policy.
 
     Args:
-      policy_path: the path to the policy directory to collect data with.
+      policy: an policy_saver.Policy object to collect data with.
```
Comment (mtrofin) [inline compiler_opt/rl/local_data_collector.py:124]: nit: typo (a policy_saver...)

**Comment by mtrofin** [comment]
lgtm. (The naming is now confusing indeed, ack on the future refactoring)

**Pull request merged**